client/middleware/api-middleware.js

```diff
@@ -58,6 +66,73 @@ export default function callAPIMiddleware({ dispatch, getState }) {
                 return ret;
             },
             error => {
+                if(error.status === 401) {
```

Comment:
The code here is a little messy and repetitive and is definitely an area that could be improved